pkg/controllers/cloud_config_sync_controller_test.go

```diff
@@ -164,30 +173,17 @@ var _ = Describe("Cloud config sync controller", func() {
 			GracePeriodSeconds: pointer.Int64(0),
 		}
 
-		if infraCloudConfig != nil {
-			Expect(cl.Delete(ctx, infraCloudConfig, deleteOptions)).To(Succeed())
-			Eventually(
-				apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(infraCloudConfig), &corev1.ConfigMap{})),
-			).Should(BeTrue())
-		}
-
-		if managedCloudConfig != nil {
-			Expect(cl.Delete(ctx, managedCloudConfig, deleteOptions)).To(Succeed())
-			Eventually(
-				apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(managedCloudConfig), &corev1.ConfigMap{})),
-			).Should(BeTrue())
-		}
-
-		if syncedCloudConfigMap != nil {
-			Expect(cl.Delete(ctx, syncedCloudConfigMap, deleteOptions)).To(Succeed())
+		allCMs := &corev1.ConfigMapList{}
+		Expect(cl.List(ctx, allCMs)).To(Succeed())
+		for _, cm := range allCMs.Items {
+			Expect(cl.Delete(ctx, &cm, deleteOptions)).To(Succeed())
```

Comment:
It's bad practice to take the location of a range variable, I'd recommend doing a `cm.DeepCopy()` here instead, same on line 181